### PR TITLE
Add tests for virtualenvwrapper, pyenv-virtuaenv, conda and a fixes

### PR DESCRIPTION
**Pull request by DonJayamanne**
A lot of fixes/tweeks as a result of the tests finding issues.

### CODE REVIEW

File: crates/pet-conda/src/environment_locations.rs [hunk 1]
```diff
@@ -117,6 +117,10 @@ pub fn get_environments(conda_dir: &Path) -> Vec<PathBuf> {
                     .collect(),
             );
         }
+        // Then read the .condarc in the conda install folder as well.
+        if let Some(mut conda_rc) = Condarc::from_path(conda_dir) {
```
Comment (DonJayamanne) [inline crates/pet-conda/src/environment_locations.rs:121]: Missed this earlier

File: crates/pet-conda/src/manager.rs [hunk 2]
```diff
@@ -59,32 +58,6 @@ pub fn find_conda_binary_on_path(env_vars: &EnvVariables) -> Option<PathBuf> {
...
-                    return Some(conda_path);
-                }
-            }
-        }
-    }
-    None
-}
-
-/// Find the conda binary on the system
-pub fn find_conda_binary(env_vars: &EnvVariables) -> Option<PathBuf> {
```
Comment (DonJayamanne) [inline crates/pet-conda/src/manager.rs:80]: Not required (old code)

File: crates/pet-virtualenvwrapper/src/lib.rs [hunk 3]
```diff
@@ -33,8 +33,10 @@ impl Locator for VirtualEnvWrapper {
             return None;
         }
         let mut name = None;
-        if let Some(filename) = &env.prefix {
-            name = filename.to_str().map(|f| f.to_string());
+        if let Some(prefix) = &env.prefix {
+            if let Some(filename) = prefix.file_name() {
```
Comment (DonJayamanne) [inline crates/pet-virtualenvwrapper/src/lib.rs:37]: Bug fix

File: crates/pet/tests/ci_test.rs [hunk 4]
```diff
@@ -52,6 +52,73 @@ fn verify_validity_of_discovered_envs() {
...
+
+    let environments = reporter
+        .reported_environments
+        .lock()
+        .unwrap()
+        .clone()
+        .into_values()
+        .collect::<Vec<_>>();
+
+    assert!(
```
Comment (DonJayamanne) [inline crates/pet/tests/ci_test.rs:76]: More tests

File: crates/pet/tests/ci_test.rs [hunk 5]
```diff
@@ -52,6 +52,73 @@ fn verify_validity_of_discovered_envs() {
...
+        environments.iter().any(
+            |env| env.category == PythonEnvironmentCategory::PyenvVirtualEnv
```
Comment (DonJayamanne) [inline crates/pet/tests/ci_test.rs:111]: more tests, and fixed issues found while adding this test

File: crates/pet-conda/src/environment_locations.rs [hunk 6]
```diff
@@ -254,51 +257,8 @@ pub fn get_known_conda_install_locations(env_vars: &EnvVariables) -> Vec<PathBuf
...
     known_paths
 }
-
-#[cfg(windows)]
-pub fn get_known_conda_locations(env_vars: &EnvVariables) -> Vec<PathBuf> {
-    let user_profile = env_vars.userprofile.clone().unwrap_or_default();
-    let program_data = env_vars.programdata.clone().unwrap_or_default();
-    let all_user_profile = env_vars.allusersprofile.clone().unwrap_or_default();
-    let home_drive = env_vars.homedrive.clone().unwrap_or_default();
-    let mut known_paths = vec![
```
Comment (DonJayamanne) [inline crates/pet-conda/src/environment_locations.rs:270]: unnecessary, duplicate code removed

**Review by karthiknadig — APPROVED** [review]

**Pull request merged**